app/controllers/admin/products_controller.rb

```diff
@@ -9,9 +9,15 @@ class ProductsController < ::Admin::AdminController
 
       def index
         begin
-          products = ::Stripe::Product.list
+          product_ids = Product.all.pluck(:external_id)
+          products = []
 
-          render_json_dump products.data
+          if product_ids.present?
+            products = ::Stripe::Product.list({ ids: product_ids })
+            products = products[:data]
```

Comment:
Should we have else ` ::Stripe::Product.list` here? 

This is tricky and I am thinking about backward compatibility. What if someone is already using that plugin, and update to the latest version? So because everything was set up before, they will have no Products in DB and as a result, nothing will be returned here